src/osdc/ObjectCacher.cc

```diff
@@ -1156,6 +1159,13 @@ void ObjectCacher::bh_write_commit(int64_t poolid, sobject_t oid,
 		       << dendl;
       }
     }
+
+    while (!hit.empty()) {
+      BufferHead *bh = hit.front();
+      hit.pop_front();
+      if (bh)
```

Comment:
#11262 . In hit, the first may merged w/ the second. So i need know which be merged(delete). Yes as you said, set the local var to nullptr after delete don't indicate the related var in hit. But change the merge_left right to BufferHead **right. I think it's not a goog method.